File: frontend/src/types/notifications.ts
export type NotificationType = 'emergency' | 'warning' | 'information' | 'system';
export type NotificationSource = 'official' | 'community' | 'system' | 'derived';

export type FloodGuardNotification = {
  id: string;
  type: NotificationType;
  title: string;
  message: string;
  location_label?: string | null;
  source: NotificationSource;
  is_read: boolean;
  requires_action: boolean;
  created_at: string;
  updated_at?: string | null;
  expires_at?: string | null;
  is_demo: boolean;
};

export type NotificationListResponse = {
  items: FloodGuardNotification[];
  next_cursor?: string | null;
  total: number;
  is_demo: boolean;
};

File: backend/app/schemas/notifications.py
from enum import Enum

from pydantic import BaseModel, Field


class NotificationType(str, Enum):
    EMERGENCY = 'emergency'
    WARNING = 'warning'
    INFORMATION = 'information'
    SYSTEM = 'system'


class NotificationSource(str, Enum):
    OFFICIAL = 'official'
    COMMUNITY = 'community'
    SYSTEM = 'system'
    DERIVED = 'derived'


class Notification(BaseModel):
    id: str
    type: NotificationType
    title: str
    message: str
    location_label: str | None = None
    source: NotificationSource
    is_read: bool = False
    requires_action: bool = False
    created_at: str
    updated_at: str | None = None
    expires_at: str | None = None
    is_demo: bool = False


class NotificationListResponse(BaseModel):
    items: list[Notification] = Field(default_factory=list)
    next_cursor: str | None = None
    total: int
    is_demo: bool = False
